src/services/users/utils.hooks.js

```diff
@@ -50,4 +52,29 @@ function preventChangesIfNotOwner(field) {
     };
 }
 
-module.exports = { checkAdminOrOwner, checkOwner, preventChangesIfNotOwner };
+async function checkOldPassword(context) {
+    const oldPassword = context.data.oldPassword;
+    const user = context.params.user;
+
+    if (context.data.hasOwnProperty('password')) {
+        if (!context.data.hasOwnProperty('oldPassword') || oldPassword === '') {
+            throw new BadRequest('Old password is required.');
+        } else {
+            try {
+                const verifier = new Verifier(context.app, { service: 'users', passwordField: 'password' });
+                await verifier._comparePassword(user, oldPassword);
```

Comment:
This is a yellow flag. We should not be using private methods (`_`), but we will leave it for now.